docs/design/storage-auto-scale.md

```diff
@@ -0,0 +1,95 @@
+# Storage Automatic Scaler
+
+## Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices OSDs so this feature helps in automatic scaling. 
+
+## Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+## High-Level Design
+
+- Create a StorageAutoScale CR per Storagecluster device class
+    ```
+    apiVersion: ocs.openshift.io/v1
+    Kind: StorageAutoScaling
+    Metadata:
+        name: <storageclusterName>-<deviceClassName>
+        namespace: openshift-storage
+    Spec:
+        storageClusterName: <storageclusterName>
+        deviceClassName: <deviceClassName>
+        capacityLimit: <capacity>
```

Comment:
Capacity is a loaded word. It might mean available capacity provided by the device set, or storage capacity of the ODS. I would prefer us using a more specific term.

Reply:
Agreed. This could be total usable Ceph capacity, aggregate raw disk capacity, or per-disk capacity. We should make sure our API makes it clear what scenario it is.

Reply:
Updated to use `clusterCapacityLimit`

Reply:
This is still somewhat unclear to me. Is this total capacity of all disks, raw? Is this the Ceph cluster capacity after factoring in replication/erasure-coding?

Reply:
@BlaineEXE  It is the total capacity of all disks, what the final size disk can grow horizontally and vertically,
 
 size of all  OSD per device class* max size <= clusterCapacityLimit

As it is already a CR specific to device class, it was fine to not to add a prefix to it `deviceClassStorageCapacityLimit`
But I can change it to `deviceClassStorageCapacityLimit` unless any other  suggestions